src/types/index.ts

```diff
@@ -82,7 +82,16 @@ export interface CommonOptions {
   closeButtonClassName?: string | string[];
 }
 
-export interface PluginOptions extends CommonOptions {
+type ToastTypeDefaults = {
+  /**
+   *  Default toast options indexed by type of the toast.
+   *
+   *  Key can be any of `success error default info warning`
+   */
+  [key in TYPE]?: ToastOptions;
```

Comment:
This is great, but I think that having all of the types available on the root of PluginOptions does not feel right.
Perhaps there could be a `toastDefaults` key holding them?

```ts
type ToastTypeDefaults = {
  toastDefaults?: {
    [key in TYPE]?: ToastOptions;
  };
};
```
Or even leave `ToastTypeDefaults ` as it is and instead of extending `PluginOptions`, create a `toastDefaults` on it that uses `ToastTypeDefaults` as the type:
```ts
export interface PluginOptions extends CommonOptions {
  // ...
  toastDefaults?: ToastTypeDefaults;
};
```